<!doctype html>
<html>

<head>
	<meta charset="utf-8" />
	<title>Using ESM to Predict Flu Strains</title>
	<link rel="stylesheet" href="reveal.js/dist/reveal.css" />
	<link rel="stylesheet" href="reveal.js/dist/theme/white.css" />
	<link rel="stylesheet" href="custom.css">
	<script src="plugin/math/math.js"></script>
	<link rel="stylesheet" href="plugin/math/math.css">

</head>

<style>
	.fit-image {
		max-width: 100%;
		max-height: 100%;
	}
</style>

<body>
	<div class="reveal">
		<div class="slides">

			<!--<section data-markdown>
				<textarea data-template>
    ## Slide 1
    A paragraph with some text and a [link](https://hakim.se).
  </textarea>
			</section>-->

			<section style="position: relative; padding-top: 2em;">
				<h4>Using ESM to Predict Viable Flu Strains</h4>
				<div style="padding-top: 2em;"></div>

				<!-- Author name -->
				<p style="color: rgb(58, 58, 137); margin: 0;">
					Carlos Avendaño
				</p>
				<!-- Date, smaller and muted -->
				<p style="font-size: 0.75em; color: gray; margin: 0;">
					June 11, 2025
				</p>
			</section>

			<section>
				<h4>Influenza is constantly evolving</h4>
				<div style="padding-top: 2em;"></div>

				<p class="fragment" Fade in> Each year, public health organizations and researchers must develop a new
					vaccine.</p>

				<div style="padding-top: .25em;"></div>

				<p class="fragment" Fade in> To produce an effective vaccine, forecasting models are used to predict
					future viral variants.</p>
			</section>


			<section>
				<!-- Title fixed at top -->
				<h4 style="position: absolute;
             top: 0;
             left: 0;
             width: 100%;
             text-align: center;
             margin: 0;">
					Why use ESM for Flu Prediction?
				</h4>

				<!-- Give your content room below the title -->
				<div style="padding-top: 3em;">
					<p>
						ESM and other large language models may be effective tools to assist in predicting next
						season’s
						flu strains more accurately, by uncovering new patterns in the proteins of successful flu
						variants.
					</p>
				</div>
			</section>


			<section style="position: relative; padding-top: 3em;">
				<h4 style="position: absolute; top: 0; left: 0; right: 0; text-align: center; margin: 0;">
					What is ESM?
				</h4>

				<p style="margin-bottom: 1.5em;">
					Evolutionary Scale Modeling (ESM) is a large language model developed by Meta, which can predict
					protein viability, structure and function.
				</p>

				<p style="margin-bottom: 1.5em; font-size: 0.70em; color: rgb(0, 0, 0); margin: 0; font-style: italic;">
					In this project <strong>ESM2</strong> was used, but will be referred to as <strong>ESM</strong>.
				</p>

				<p style="position: absolute; bottom: -4em; right: 1em; font-size: 0.6em; color: gray;">
					<a href="https://www.science.org/doi/10.1126/science.ade2574" target="_blank"
						rel="noopener noreferrer">
						Meta AI, Science (2022)
					</a>
				</p>
			</section>




			<section>
				<h4>PRETRAINING</h4>

				<div style="padding-top: 2em;"></div>

				<p>Machine learning models are fed huge amounts of data to infer associations, for ESM the largest
					model was trained on 15 billion proteins from uniparc.</p>

			</section>

			<section data-auto-animate>
				<h4>Three Base Pretrained Models</h4>
				<p style="margin-bottom: 2em;"></p>
			</section>

			<section data-auto-animate>
				<h4>Three Base Pretrained Models</h4>
				<p style="margin-bottom: 2em;"></p>
				<ul>
					<li>650 Million Protein Model</li>
					<li>3 Billion Protein Model</li>
					<li>15 Billion Protein Model</li>
				</ul>
			</section>

			<section>
				<h5>Larger Models Are More Computationally Intensive to Run</h5>

				<p style="margin-bottom: 1em;"></p>

				<img src="images/runtime_esm2_base_mdls.png"
					style="height: 500px; margin: 0 auto 4rem auto; background: transparent;"
					alt="Runtime for ESM2 Base Models">
			</section>

			<section>
				<h4>FINE-TUNING</h4>

				<p>These models can be further refined and trained on specific data.</p>
				<img src="images/Alignment.png" alt="Alignment">

			</section>

			<section>
				<h4>Nextstrain Tree</h4>
				<img src="images/nextstrain_60y_flu_tree.png" alt="Tree">
				<p style="font-size: 0.55em;">60 year Nextstrain Tree for H3N2 (for flu segment HA).</p>
			</section>
			</section>

			<section>
				<h4>Snakemake Pipeline</h4>
				<img src="images/Workflow.png" alt="Pipeline">
			</section>

			<section>
				<h4>Translate root sequence nucleotide FASTA to amino acids</h4>
				<p style="margin-bottom: 1em;"></p>
				<pre
					style="font-size: 0.75em; line-height: 1; max-height: 70vh; overflow: auto; background: #f5f5f5; padding: .5em; border-radius: 2px;">
					<code>
{
"nuc": "CAAACCATTTGAATGGATGTCAATCCGACT",
"PB1": "MDVNPTLLFLKVPAQNAISTTFPYTGDPPY"
}
					</code>
				</pre>

				<p style="font-size: 0.75em;">ESM is fed protein sequences, for each flu segment, translate the root
					sequences from nucleotide to
					amino acid.</p>
			</section>

			<section>
				<h4>Extract FASTAs for each node in Nextstrain tree</h4>
				<p style="margin-bottom: 1em;"></p>
				<pre
					style="font-size: 0.75em; line-height: 1; max-height: 30vh; overflow: auto; background: #f5f5f5; padding: 0em; border-radius: 0px;">
					<code>
"branch_attrs": {
	"mutations": {
	"nuc": [
		"C1665T"
	]
	}
}
					</code>
				</pre>

				<p style="font-size: 0.75em;">Nextstrain doesn't store full sequences for every node, so we use the root
					sequence and the mutations to reconstruct each sequence.</p>
			</section>

			<section>
				<h4>Calculate maximum frequency for each node in Nextstrain tree</h4>
			</section>

			<section>
				<h4>Run sequences through ESM</h4>

				<p style="margin-bottom: 1em;"></p>

				<p style="font-size: 0.75em;">Compute log likelihood for extracted sequences for each flu segment, using
					each base ESM model.</p>
			</section>

			<section>
				<h4>Extract sequences for model fine‑tuning</h4>
				<p style="margin-bottom: 1em;"></p>
				<p style="font-size: 0.75em;">Option to include a cuttoff for sequences before a given time frame.</p>
			</section>

			<section>
				<h4>Generate fine‑tune models</h4>
				<p style="margin-bottom: 1em;"></p>
				<p style="font-size: 0.75em;">Feed extracted fastas into ESM</p>
			</section>

			<section>
				<h4>Run sequences through ESM fine‑tuned models</h4>
				<p style="margin-bottom: 1em;"></p>
				<p style="font-size: 0.75em;">Calculate log likelihood</p>
			</section>

			<section>
				<h4>Nextstrain Tree with Log Likelihood</h4>
				<img src="images/Nextstrain_ESM_Scores.svg" alt="Nextstrain ESM Scores">
			</section>

			<section>
				<h4>Fine Tune Vs Base 650M Model for HA </h4>
				<img src="images/HA_650M_v_650M_TN_Scatterplots.png"
					style="height: 650px; margin: 0 auto 4rem auto; background: transparent;"
					alt="Fine Tune vs Base 650M Model for HA">
			</section>

			<section>
				<h5>Fine Tune vs Base 650M ESM Model Log Likelihood Over Time</h5>
				<img id="comparison-image" src="images/650M_4x4_grid.png" style="width: 800px;">
			</section>


			<section>
				<h5>Fine Tune vs Base 3B ESM Model Log Likelihood Over Time</h5>
				<img src="images/3B_4x4_grid.png">
			</section>

			<section>
				<h5>ESM Log Likelihood vs Maximum Frequency Base Models Compared</h5>
				<img src="images/HA_LL_vs_Max_Frequency.png">
			</section>

			<section>
				<h5>Spearman Correlation Comparison Base Model Compared Grouped</h5>
				<img src="images/Spearman_Correlation_Comparision_Base_Model.png"
					alt="Spearman Correlation Comparison Base Model">
			</section>

			<section>
				<h5>ESM Log Likelihood vs Maximum Frequency Fine Tune Models Compared</h5>
				<img src="images/HA_LL_vs_Max_Frequency_Fine_Tune_650M.png">
			</section>

			<section>
				<h5>Comparing Fine Tune Models to Base Models Grouped</h5>
				<div style="width: 100%; height: 100%; display: flex; justify-content: center; align-items: center;">
					<img src="images/Combined_Spearman_Fine_Tune_Comparison.png"
						style="height: 650px; margin: 0 auto 4rem auto; background: transparent;">
				</div>
			</section>

			<section>
				<h5>Adjusting Fine Tuning Parameters</h5>
				<div style="height: 650px; display: flex; justify-content: center; align-items: center;">
					<img src="images/Spearman_Correlation_Comparison_Fine_Tune_time_range.png">
				</div>
			</section>

			<section>
				<h5>Adjusting Fine Tuning Parameters</h5>
				<div style="width: 100%; height: 100%; display: flex; justify-content: center; align-items: center;">
					<img src="images/Spearman_Correlation_Comparison_All_Models.png">
				</div>
			</section>

			<section>
				<h5>Time Series Cross Validation Expected</h5>
				<img src="images/Spearman_Correlation_Comparison_Fine_Tune_time_series_cross_validation_dummy.png"
					style="margin: 0 auto 4rem auto; background: transparent;">
			</section>
			<section>
				<h5>Time Series Cross Validation Actual</h5>
				<img src="images/Spearman_Correlation_Comparison_Fine_Tune_time_series_cross_validation.png"
					style="margin: 0 auto 4rem auto; background: transparent;">
			</section>

			<section>
				<h5>Working With A larger flu tree</h5>
				<img src="images/nextstrain_lg_tree.png" style="margin: 0 auto 4rem auto; background: transparent;">
			</section>

			<section>
				<h5>Working With A larger flu tree training model up to 1990</h5>
				<img src="images/Spearman_Correlation_Comparison_LG_Tree_1990.png"
					style="margin: 0 auto 4rem auto; background: transparent;">
			</section>

			<section>
				<h5>Spearman Correlation Large and Small trees, 1990 to 2005</h5>
				<img src="images/Spearman_Correlation_Overlay_1990_to_2005.png"
					style="margin: 0 auto 4rem auto; background: transparent;">
			</section>

			<section>
				<h5>Next Steps</h5>
				<ul>
					<li>Compare the ability of ESM vs DMS to forecast evolutionarily successful influenza viruses.</li>
					<li>Train models on larger amounts of sequences to prevent over-training in fine tuning and run ESM
						on other viruses: SARS-CoV-2</li>
				</ul>
			</section>

			<section>
				<h3>Thank You</h3>
			</section>

			<!-- Reveal.js core and plugins -->
			<script src="reveal.js/dist/reveal.js"></script>
			<script src="reveal.js/plugin/markdown/markdown.js"></script>
			<script src="https://d3js.org/d3.v7.min.js"></script>

			<script>
				Reveal.initialize({
					hash: true,
					plugins: [RevealMarkdown]
				});
			</script>
</body>

</html>